libs/libc/machine/xtensa/arch_setjmp.S

```diff
@@ -127,8 +129,10 @@ setjmp:
 
   /* Flush registers.  */
   mov a4, a2      # save a2 (jmp_buf)
-  movi  a2, 0
-  syscall
+  movi  a2, SYS_flush_context
```

Comment:
let's reuse full_contextsave and contextrestore?

Reply:
Are any of these commands handled by the software interrupt?

Speaking of `setjmp`, please note that the current `longjmp` implementation can cause issues for code using `WINBASE` and `WINSTART` as they are not saved.  for ESP-IDF there is a patch here: https://github.com/espressif/esp-idf/blob/4350e6fef859b94f3efae6ceced751911c9d2cbc/components/esp_rom/patches/esp_rom_longjmp.S#L43-L59